### ISSUE TITLE
Update WebFrameMain.executeJavaScriptInIsolatedWorld to match WebContents interface

### ISSUE THREAD

**Issue by samuelmaddock** [electron/electron#27862]
**Preflight Checklist**

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Problem Description**

The upcoming `WebFrameMain` API implements `executeJavaScriptInIsolatedWorld` with an interface that doesn't match that of `WebContents`. `WebContents` accepts [`WebSource[]`](https://github.com/electron/electron/blob/master/docs/api/structures/web-source.md) as the second parameter while `WebFrameMain` accepts a `String`. We should try to resolve this mismatch prior to Electron v12's release on 2021-03-02.

Docs:
- [`WebFrameMain.executeJavaScriptInIsolatedWorld`](https://github.com/electron/electron/blob/master/docs/api/web-frame-main.md#frameexecutejavascriptinisolatedworldworldid-code-usergesture)
- [`WebContents.executeJavaScriptInIsolatedWorld`](https://github.com/electron/electron/blob/master/docs/api/web-contents.md#contentsexecutejavascriptinisolatedworldworldid-scripts-usergesture)

-------

`WebFrameMain`'s implementation directly calls `RenderFrameHost::ExecuteJavaScriptForTests` from the main process.
```cpp
virtual void ExecuteJavaScriptForTests(
      const base::string16& javascript,
      JavaScriptResultCallback callback,
      int32_t world_id = ISOLATED_WORLD_ID_GLOBAL) = 0;
 ```

`WebContents`'s implementation remotely calls a method on `webFrame` over a JS IPC.
https://github.com/electron/electron/blob/e406ba9558cc6a44da5fbc329ba1afee4d1679ad/lib/browser/api/web-contents.ts#L185-L188

In C++ from the renderer, this invokes the following:
https://github.com/electron/electron/blob/e406ba9558cc6a44da5fbc329ba1afee4d1679ad/shell/renderer/api/electron_api_web_frame.cc#L718-L723

The methods available from the main and renderer processes are incompatible. `WebSource` can't be used in place of a `String`.

**Proposed Solution**

Deprecate passing `WebSource[]` as a parameter to `WebContents.executeJavaScriptInIsolatedWorld`, instead only accepting code as a String.

- Based on a [cursory search](https://github.com/search?l=TypeScript&q=executeJavaScriptInIsolatedWorld&type=Code), I didn't find any examples of folks using the API to pass anything except `[{ code: '...' }]`.
- The WebContents implementation can't be used by WebFrameMain because it doesn't work without `nodeIntegrationInSubFrames`.
- Removing the ability to pass in `WebSource` means WebContents could simply call out to WebFrameMain in JS.
```js
WebContents.prototype.executeJavaScriptInIsolatedWorld = (...args) => {
   this.mainFrame.executeJavaScriptInIsolatedWorld(...args);
}
```

**Alternatives Considered**

Remove `WebFrameMain.executeJavaScriptInIsolatedWorld` prior to v12's release and re-introduce it in a minor release.

**Additional Information**

None

**Comment by nornagon** [comment]
Adding blocks-release because this is an API change and should be made before stable.

**Comment by nornagon** [comment]
I don't think we should deprecate the WebSource version. Additionally, the new API should not call `RenderFrameHost::ExecuteJavaScriptForTests`, as that method is _for tests_.

**Comment by zcbenz** [comment]
> * The WebContents implementation can't be used by WebFrameMain because it doesn't work without `nodeIntegrationInSubFrames`.

We can add a new mojo interface to implement the API with C++, without requiring node integration in renderer, and the WebFrameMain implementation would not need to use a testing API anymore.

**Comment by VerteDinde** [comment]
For the Electron 12 stable release specifically: we're going to revert this API temporarily, and then follow up with any changes in a minor release. I'll take point on the revert for 12 stable.

**Comment by smartguy1196** [comment]
#32756 - it seems that the executeJavaScript method on webFrameMain needs to be reworked altogether

**Comment by samuelmaddock** [comment]
May be solved by #35476
